sdk/core/core/src/az_http_policy_logging.c

```diff
@@ -20,63 +20,97 @@ enum
         // that they don't blow up the logs.
 };
 
-static az_result _az_http_policy_logging_append_lengthy_value(az_span value, az_span* ref_log_msg)
+static az_span _az_http_policy_logging_append_lengthy_value(az_span ref_log_msg, az_span value)
 {
+  // The caller should validate that ref_log_msg is large enough to contain the value az_span
+  // This means, ref_log_msg must have at least _az_LOG_LENGTHY_VALUE_MAX_LENGTH (i.e. 50) bytes
```

Comment:
Where did 50 come from?

Reply:
That's what `_az_LOG_LENGTHY_VALUE_MAX_LENGTH` is set to. I don't know exactly what the rationale was to pick this number. If I had to guess, it might be based on what we want as a single line length to be written to the console, so anything larger than 50 would get written out as (or it's arbitrarily chosen as a small enough value):
"some content .... trimmed end"

This was already defined here:
https://github.com/Azure/azure-sdk-for-c/blob/c94c83ba5eb4733b2545abf7ed8f9e20819c021d/sdk/core/core/src/az_http_policy_logging.c#L14-L21

@antkmsft wrote this code, so he can probably answer that better.